src/Language/DifferentialDatalog/Syntax.hs

```diff
@@ -590,8 +590,32 @@ instance Show Atom where
 -- positive/negative polarity, Boolean conditions, aggregation,
 -- disaggregation (flatmap), inspect operations.
 data RuleRHS = RHSLiteral   {rhsPolarity :: Bool, rhsAtom :: Atom}
-             | RHSCondition {rhsExpr :: Expr}
-             | RHSAggregate {rhsVar :: String, rhsGroupBy :: Expr, rhsAggFunc :: String, rhsAggExpr :: Expr}
+               -- Expression that can filter or map input collation, or both:
+               -- * Filtering: `x == y`
+               -- * Mapping: `var x = f(y)`
+               -- * Filter/map: `Some{x} = f(y)`
+             | RHSCondition {
```

Comment:
is this then a good name?

Reply:
Nope.  But I cannot think of a better name. Filter is probably less confusing.